File: app/src/main/java/com/biginsect/easygithub/ui/activity/SplashActivity.java
package com.biginsect.easygithub.ui.activity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.biginsect.easygithub.ui.base.BaseActivity;
import com.biginsect.easygithub.ui.contract.ISplashContract;
import com.biginsect.easygithub.ui.presenter.SplashPresenter;


/**
 * app启动之后首先启动的页面，为了防止出现白屏
 *@author biginsect
 */

public class SplashActivity extends BaseActivity<ISplashContract.ISplashView, ISplashContract.ISplashPresenter>
        implements ISplashContract.ISplashView{

    @Override
    protected int getLayoutId() {
        return 0;
    }

    @NonNull
    @Override
    public ISplashContract.ISplashPresenter createPresenter() {
        presenter = new SplashPresenter();
        return presenter;
    }

    @Override
    protected void initActivity() {
        super.initActivity();
        presenter.getUser();
    }

    @Override
    public void showHomePage() {
        finishDelay();
        Uri uri = getIntent().getData();
        if (null != uri) {
            startActivity(new Intent(getActivity(), HomePageActivity.class));
        }
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        super.initView(savedInstanceState);
    }
}

File: app/src/main/java/com/biginsect/easyhub/ui/activity/SplashActivity.kt
package com.biginsect.easyhub.ui.activity

import android.content.Intent
import android.os.Bundle
import com.biginsect.easyhub.ui.base.BaseActivity
import com.biginsect.easyhub.ui.contract.ISplashContract
import com.biginsect.easyhub.ui.presenter.SplashPresenter

/**
 * @author big insect
 * @date 2018/8/30.
 */
class SplashActivity :BaseActivity<ISplashContract.ISplashView, ISplashContract.ISplashPresenter>(),
        ISplashContract.ISplashView{

    override fun showHomePage() {
        finishDelay()
        val uri = intent.data
        if(null == uri){
            startActivity(Intent(activity, HomePageActivity::class.java))
        }
    }

    override fun createPresenter(): ISplashContract.ISplashPresenter {
        presenter = SplashPresenter()
        return presenter
    }

    override fun getLayoutId(): Int {
        return 0
    }

    override fun initActivity() {
        super.initActivity()
        presenter.getUser()
    }

    override fun initView(savedInstanceState: Bundle?) {
        super.initView(savedInstanceState)
    }

    override fun showLoginPage() {
        finishDelay()
        startActivity(Intent(this, LoginActivity::class.java))
    }
}